gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -0,0 +1,172 @@
+            }
+        }
+
+        return null;
+    }
+
+    private DateTimeType getOccurrenceDate(IVLTS effectiveTime) {
+        if (effectiveTime != null && effectiveTime.getCenter() != null) {
+            return DateFormatUtil.parse(effectiveTime.getCenter().getValue());
+        }
+
+        return null;
+    }
+
+    private Reference getAgentReference(List<RCMRMT030101UK04Participant> participantList, RCMRMT030101UK04EhrExtract ehrExtract,
+                                        II planStatementID) {
+        Reference reference = new Reference();
+        var nonNullFlavorParticipants = participantList.stream()
+                .filter(participant -> isNotNullFlavour(participant))
+                .collect(Collectors.toList());
+
+        var pprfParticipants = getParticipantReference(nonNullFlavorParticipants, PPRF_PERFORMER);
+        if (pprfParticipants.isPresent()) {
+            return reference.setReference(pprfParticipants.get());
+        }
+
+        var prfParticipants = getParticipantReference(nonNullFlavorParticipants, PRF_PERFORMER);
+        if (prfParticipants.isPresent()) {
+            return reference.setReference(prfParticipants.get());
+        }
+
+        var ehrComposition = EhrResourceExtractorUtil.extractEhrCompositionForPlanStatement(ehrExtract, planStatementID);
+        var participant2Reference = getParticipant2Reference(ehrComposition);
+        if (participant2Reference.isPresent()) {
+            return reference.setReference(participant2Reference.get());
+        }
+
+        // TODO: if none of these are present, then we should reference an 'Unknown User' Practitioner (NIAD-2026)
+
+        return reference;
+    }
+
+    private Optional<String> getParticipantReference(List<RCMRMT030101UK04Participant> participantList, String typeCode) {
+        return participantList.stream()
+                .filter(participant -> hasTypeCode(participant, typeCode))
+                .filter(participant -> hasAgentReference(participant))
+                .map(RCMRMT030101UK04Participant::getAgentRef)
+                .map(RCMRMT030101UK04AgentRef::getId)
+                .map(II::getRoot)
+                .findFirst();
+    }
+
+    private Optional<String> getParticipant2Reference(RCMRMT030101UK04EhrComposition ehrComposition) {
+        return ehrComposition.getParticipant2().stream()
+                .filter(participant2 -> participant2.getNullFlavor() == null)
+                .map(RCMRMT030101UK04Participant2::getAgentRef)
+                .map(RCMRMT030101UK04AgentRef::getId)
+                .map(II::getRoot)
+                .findFirst();
+    }
+
+    private boolean hasAgentReference(RCMRMT030101UK04Participant participant) {
+        return participant.getAgentRef() != null && participant.getAgentRef().getId() != null;
+    }
+
+    private boolean hasTypeCode(RCMRMT030101UK04Participant participant, String typeCode) {
+        return participant.getTypeCode().get(0).equals(typeCode);
```

Comment:
I think you need to check if `participant.getTypeCode()` is not empty before taking the first element (it will throw 
IndexOutOfBoundsException if the list is empty)